auth/cloudfrontAuthFunction.js

```diff
@@ -0,0 +1,40 @@
+// This is a CloudFront Viewer Request function that puts a simple authentication layer in from ouf our .ics files.
+// It expects a ?key be present in the URL for any file behind the CloudFront distribution, for example:
+// https://api.discal.app/guilds/890053994926460939/events.ics?key=XXXX
+
+// Any request that doesn't have a Guild ID in the expected part of the URL, or has an invalid key, will receive a 401 response.
+
+// This means nothing but these .ics files - in our /guilds/<id>/etc format - can ever be served from the domain the CF distro serves, but that's fine.
+
+// This code is deployed through a weird <textarea> in CloudFront > Functions.
+
+// See generateApiKeyForGuild.js for details on generated the key for any given Guild.
+
+var crypto = require('crypto');
+
+var response401 = {
+  statusCode: 401,
+  statusDescription: 'Unauthorized'
+};    
+
+function handler(event) {
+    if(!event.request.querystring.key){
+      return response401;
+    }
+    
+     // We can't setup this key with an env var or secret or anything, and i'm not going to commit it here.
+     // So before modifying this in CloudFront you'll have to enter the real key, which is DISCAL_AUTH_SECRET in .env.
+    var defcalSecret = "XXXXXXXX"
+    
+    // Expects the URI to be like ""https://discal.app/guilds/1232354345345/..."
+    // We want to extract that ID.
+    var guildId = event.request.uri.match(/guilds\/(.*)\//)[1].toUpperCase() // Upcase it here, and when we generate the key, just to avoid any case sensitivity foolishness.
```

Comment:
Maybe worth using `(\d+)` instead of `(.*)` since we're pretty certain it'll be a numeric ID?

Reply:
I did think about restricting it to numbers, but yeah if you put in a bad ID it'll fail anyway. 

Also, I have a dim memory of having to update a database scheme when someone's API migrated their IDs from ints to alphanumerics, and I don't want to go back and change reg exp if Discord every does the same.